daemon/logger/awslogs/cloudwatchlogs.go

```diff
@@ -221,6 +254,21 @@ func (l *logStream) create() error {
 	return err
 }
 
+// create creates log group and log stream for the instance of the awslogs logging driver
+func (l *logStream) create() error {
```

Comment:
My opinions on both items raised by @samuelkarp:


> would it be better to attempt calling CreateLogStream first and only call CreateLogGroup if there is an error because the group doesn't exist? I think I would prefer the second option, since it won't require that you have the logs:CreateLogGroup permission in your IAM policy as long as the group already exists.

This one gets my vote.

> The log stream allows templates. Should the log group also allow templates? I don't have a strong opinion here.

I think templates should definitely be supported for log groups.